YoutubeDownloader/ViewModels/Components/DashboardViewModel.cs

```diff
@@ -274,8 +276,9 @@ or DownloadStatus.Canceled
         }
     }
 
-    public void RestartDownload(DownloadViewModel download)
+    public void RestartDownload(object parameter)
```

Comment:
Is there no way to do the type casting on the view side? So as to keep `DownloadVIewModel download` here instead.

Reply:
unfortunately not, as Avalonias binding only binds to methods with no or object parameter.
But we could use the MVVM toolkit's sourcegenerator to create a command for the method with the generic parameter (just adding a RelayCommand attribute on the method), then the UI could bind to the command and this method would not be required.

Reply:
Oh and I assume then it won't need to leverage reflection (assuming it does now)? I think that sounds like a good idea, can we try to do that? If it's not too much effort.

Reply:
I am not sure if binding to commands uses reflection, but general I think Binding uses reflection in every XAML framework. Using CompiledBindings would not use reflections, but there seems to be a bug or something I don't understand why it's not working with compiled bindings enabeld, will investigate after that PR to enable compiled bindings. But can change to using commands now, it is not much effort.